### PR TITLE
Restore some X509 functions to public headers

### PR DESCRIPTION
**Pull request by davidhorstmann-arm**
Restore some X509 functions that were erroneously removed in #8715 to public headers.

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** not needed
- [x] **backport** not required - functions were only made private in 3.x
- [x] **tests** not required - no functional changes, only to the header location of functions

### CODE REVIEW

**Review by tom-daubney-arm — DISMISSED** [review]
Subject to CI passing, this LGTM.

**Comment by tom-daubney-arm** [comment]
OpenCI happy so I'm removing the label

File: ChangeLog.d/restore-x509-functions-to-public.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Bugfix
```
Comment (mpg) [inline ChangeLog.d/restore-x509-functions-to-public.txt:1]: Thanks for not forgetting the ChangeLog, but this time I don't think an entry is needed, considering the bug was never present in any released version. (And the ChangeLog is about changes between say 3.5.2 and 3.6.0, not transient states in `development` in the meantime.)

File: include/mbedtls/x509.h [hunk 2]
```diff
@@ -324,6 +324,133 @@ int mbedtls_x509_dn_gets(char *buf, size_t size, const mbedtls_x509_name *dn);
 
 int mbedtls_x509_string_to_names(mbedtls_asn1_named_data **head, const char *name);
```
Comment (mpg) [inline include/mbedtls/x509.h:325]: Pre-existing: I think this one should have been removed by #8715. It was below the "Internal module functions" marker and has no doxygen doc. Sorry for noticing only now, but could you make it internal too?

**Review by mpg — CHANGES_REQUESTED** [review]
Thanks for fixing our mistakes :)

This time I was more careful and checked the diff with 3.5.2 to make sure no function was removed that wasn't already marked internal. I'm happy on this front, but this made me notice an function that was marked internal has actually been left public.

**Review by mpg — DISMISSED** [review]
LGTM, thanks!

**Review by tom-daubney-arm — APPROVED** [review]
LGTM

**Comment by gilles-peskine-arm** [comment]
The CI is not happy.

**Comment by davidhorstmann-arm** [comment]
> Pre-existing: I think this one should have been removed by #8715. It was below the "Internal module functions" marker and has no doxygen doc.

@mpg It seems that `mbedtls_x509_string_to_names()` was not moved to private because it was used in `programs/x509/cert_req.c` (which is causing CI failures). Given that it is the direct reverse of `mbedtls_x509_dn_gets()`, which is public, would it be okay to make this public too?

Would you accept adding a docstring to this function and keeping it in the public header?

**Comment by mpg** [comment]
@davidhorstmann-arm Ah, so it wasn't missed, there was a reason Valerio kept it. Yes, I agree with make it public with an added docstring. Thanks!

**Review by mpg — APPROVED** [review]
LGTM, thanks!

**Review by tom-daubney-arm — APPROVED** [review]
Assuming CI success - this LGTM

**Pull request merged**